Error installing in a clean project

## Summary
I have created a new completely clean project the only thing I have installed is react-native-maps with the following command:
```
npm install react-native-maps --save-exact
```

## Environment info
`react-native info` output:

```bash
System:
    OS: macOS 12.2.1
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Memory: 211.51 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.8.0 - /usr/local/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 8.6.0 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7042882
    Xcode: 13.3/13E113 - /usr/bin/xcodebuild
  Languages:
    Java: javac 17 - /usr/local/opt/openjdk/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.0 => 0.68.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

```

## Steps to reproduce

1.  ```react-native init test```
2. ```cd test```
3. ```npm install react-native-maps --save-exact```

Result:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: test@0.0.1
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"17.0.2" from the root project
npm ERR!   peer react@">= 16.0 || < 18.0" from react-native-maps@0.30.1
npm ERR!   node_modules/react-native-maps
npm ERR!     react-native-maps@"*" from the root project
npm ERR!   2 more (react-native, react-native-web)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.0.0" from react-dom@18.0.0
npm ERR! node_modules/react-dom
npm ERR!   peer react-dom@">=17.0.1" from react-native-web@0.17.7
npm ERR!   node_modules/react-native-web
npm ERR!     peer react-native-web@">= 0.11" from react-native-maps@0.30.1
npm ERR!     node_modules/react-native-maps
npm ERR!       react-native-maps@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/kkk/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kkk/.npm/_logs/2022-04-06T14_00_48_191Z-debug-0.log


```

Thanks!
Antonio

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
     "mapkit"
   ],
   "peerDependencies": {
-    "react": ">= 16.0 || < 18.0",
+    "react": ">= 16.0",
     "react-native": ">= 0.51",
     "react-native-web": ">= 0.11"
   },

